APMrover2/Log.cpp

```diff
@@ -154,6 +154,41 @@ void Rover::Log_Write_Proximity()
     DataFlash.Log_Write_Proximity(g2.proximity);
 }
 
+void Rover::Log_Write_Sail()
+{
+    // only log sail if present
+    if (!g2.motors.has_sail()) {
+        return;
+    }
+
+    // get wind direction
+    float wind_dir_abs = 0.0f;
+    float wind_dir_rel = 0.0f;
+    float true_wind_speed = 0.0f;
+    if (rover.g2.windvane.enabled()){
+        wind_dir_abs = degrees(g2.windvane.get_absolute_wind_direction_rad());
+        wind_dir_rel = degrees(g2.windvane.get_apparent_wind_direction_rad());
+        true_wind_speed = g2.windvane.get_true_wind_speed();
+    } 
+    float aparent_wind_speed = 0.0f;
+    if (rover.g2.airspeed.enabled()){
+        aparent_wind_speed = rover.g2.airspeed.get_airspeed();
+    }
+    float VMG = 0.0f;
+    if (rover._sailboat_indirect_route){
+        VMG = rover._sailboat_velocity_made_good;
+    }
+    DataFlash.Log_Write("SAIL", "TimeUS,WindDirAbs,WindDirRel,TrueWindSpd,AparentWindSpd,SailOut,VMG",
+                        "shh%", "F000", "Qfff", // not sure if i need to change this line or not??
```

Comment:
see : http://ardupilot.org/dev/docs/code-overview-adding-a-new-log-message.html#the-easy-way
first is units (https://github.com/ArduPilot/ardupilot/blob/master/libraries/DataFlash/LogStructure.h#L81), second multiplier, last format.
So it should be
"s--n%n", "F------", "Qffffff",
Standing for 
second, unknown, unknown, m/s, percent, m/s